src/stores/ClientPresenceStore.js

```diff
@@ -22,11 +22,19 @@ class ClientPresenceStore extends PresenceStore {
   async setClientPresence(presence) {
     const packet = await this._parse(presence);
     this.clientPresence.patch(packet);
-    this.client.ws.send({ op: OPCodes.STATUS_UPDATE, d: packet });
+    if (this.clientPresence.shard === 'all') {
+      this.client.ws.send({ op: OPCodes.STATUS_UPDATE, d: packet });
+    } else {
+      const shard = this.client.ws.shards.find(s => s.id === this.clientPresence.shard);
```

Comment:
Can't this be a `.shards.get()` followed by a null check?

Reply:
That'd make more sense lol